spec/model/compatibility-check/check-flex-search-on-field.spec.ts

```diff
@@ -0,0 +1,275 @@
+                    }
+                `,
+            );
+            expectSingleCompatibilityIssue(
+                result,
+                'Field "field" should enable @flexSearch (required by module "module1").',
+            );
+        });
+
+        it('accepts @flexSearch is present', () => {
+            const result = runCheck(
+                gql`
+                    type Test @rootEntity(flexSearch: true) @modules(in: "module1") {
+                        field: String @modules(all: true) @flexSearch
+                    }
+                `,
+                gql`
+                    type Test @rootEntity(flexSearch: true) {
+                        field: String @flexSearch
+                    }
+                `,
+            );
+            expectToBeValid(result);
+        });
+
+        it('accepts @flexSearch is present even though it is not required', () => {
+            const result = runCheck(
+                gql`
+                    type Test @rootEntity(flexSearch: true) @modules(in: "module1") {
+                        field: String @modules(all: true)
+                    }
+                `,
+                gql`
+                    type Test @rootEntity(flexSearch: true) {
+                        field: String @flexSearch
+                    }
+                `,
+            );
+            expectToBeValid(result);
+        });
+
+        it('rejects if @flexSearch(includeInSearch: true) is missing', () => {
+            const result = runCheck(
+                gql`
+                    type Test @rootEntity(flexSearch: true) @modules(in: "module1") {
+                        field: String @modules(all: true) @flexSearch(includeInSearch: true)
+                    }
+                `,
+                gql`
+                    type Test @rootEntity(flexSearch: true) {
+                        field: String @flexSearch
+                    }
+                `,
+            );
+            expectSingleCompatibilityIssue(
+                result,
+                'Field "field" should enable @flexSearch(includeInSearch: true) (required by module "module1").',
+            );
+        });
+
+        it('rejects if @flexSearch(includeInSearch) should be true but is false', () => {
+            const result = runCheck(
+                gql`
+                    type Test @rootEntity(flexSearch: true) @modules(in: "module1") {
+                        field: String @modules(all: true) @flexSearch(includeInSearch: true)
+                    }
+                `,
+                gql`
+                    type Test @rootEntity(flexSearch: true) {
+                        field: String @flexSearch(includeInSearch: false)
+                    }
+                `,
+            );
+            expectSingleCompatibilityIssue(
+                result,
+                'Field "field" should enable @flexSearch(includeInSearch: true) (required by module "module1").',
+            );
+        });
```

Comment:
I think it would be better to also require @suppress in this case.
There is a reason why the baseline sets a certain value so it would better to acknowledge it via @suppress.

Reply:
`false` is the default value for `includeInSearch`, so setting it to `false` explicitly also would be a kind of acknowledgement.

But then I'll keep it as-is, and I already added it to the list of questions to discuss in the review meeting.